### ISSUE TITLE
Failed to check the balance using local Vault

### ISSUE THREAD

**Issue by bochaco** [maidsafe/safe_client_libs#1002]
I just tried with v0.1.0 of CLIs (built for local Vault, i.e. non-mock-network), and all automated tests are passing with the exception of those which involve in fetching the balance from a SafeKey with `client.get_balance()`. The error I'm getting back from SCL is: 
```
Unexpected error - CoreError::Unexpected::{"No connection group found - did you call `connect`?"}
```

The easiest way to reproduce it is by authorising safe_cli and then running these two commands (note you'll need the branch from [this PR](https://github.com/maidsafe/safe-cli/pull/212) to be able to build safe_cli for real local Vault):
```
$ cargo run -- keys create
New SafeKey created at: "safe://hbyyyybdmqnaw863bn5nsz6qgm54i75m6a79nuk8g5hyqiq3bqzz43yq3e"
Key pair generated:
pk = 8d6e16287f6422d8ad7f38cbdeabddafd8efc5351cdbe01d57642ebdf5903b28d86dd38e36a6d43dc2add3fa745b27d0
sk = 41ce83f0de51bd6a4fd7de4b59e108b83b5b8bb029cd7ed37c0342c11861b568

$ cargo run -- keys balance
Enter secret key corresponding to the SafeKey to query the balance from: 41ce83f0de51bd6a4fd7de4b59e108b83b5b8bb029cd7ed37c0342c11861b568
[2019-08-22T19:45:50Z ERROR safe] safe_cli error: [Error] ContentNotFound - No SafeKey found at specified location
```

**Comment by calumcraig** [comment]
Can confirm I am seeing the same error when trying to query the balance of a key that I have just created:
```
> 
./safe keys balance --keyurl safe://hbyyyybnwqb45wxuzgmotwm34o3h69e6ekgkgrzn5xstmucqh17es571yw
Enter secret key corresponding to public key at "safe://hbyyyybnwqb45wxuzgmotwm34o3h69e6ekgkgrzn5xstmucqh17es571yw": 401903fb18f9874da4100e15997c9eb683172dce7a7ea788ecbc8fd2a2f55517
[2019-08-23T10:14:35Z ERROR safe] safe_cli error: [Error] ContentNotFound - No SafeKey found at specified location
```

**Comment by calumcraig** [comment]
There is also a failure for `safe wallet balance`:
```
> ./safe wallet balance safe://hbyyyybrhatine48hkcouttbzprn5n4xu88szkazt4z7zc4ww1e38esku6
[2019-08-23T12:23:43Z ERROR safe] safe_cli error: [Error] ContentError - Failed to read balances from Wallet: [Error] NetDataError - Failed to get MD: CoreError(No such data - CoreError::DataError -> NoSuchData)
```

**Comment by bochaco** [comment]
@nbaksalyar @calumcraig I just gave it a try with latest commit from SCL and it's looking much better, although there is still an issue when trying to check the balance of the default SafeKey, I can check the balance of other SafeKeys but when I try with the default one CLI hangs and it doesn't show the balance either, this is the only thing relevant I'm seeing on CLI log:
```
...
[2019-08-23T19:03:04Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T19:03:04Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T19:03:04Z DEBUG quic_p2p::utils] ERROR in communication with peer 127.0.0.1:35843: Connection(ApplicationClosed { reason: ApplicationClose { error_code: 0, reason: b"" } }) - Connection Error: closed by peer: 0. Details: Incoming streams failed
[2019-08-23T19:03:04Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T19:03:05Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T19:03:05Z DEBUG tokio_reactor::registration] scheduling Read for: 0
...
```

On vault log this is all I get when trying to check such SafeKey's balance:
```
INFO 2019-08-23T19:07:47Z [src/client_handler.rs:152] 13eeba..: Connected to new client on 127.0.0.1:48663
TRACE 2019-08-23T19:07:47Z [src/vault.rs:188] Node(13eeba..): Succesfully sent message to: 127.0.0.1:48663
INFO 2019-08-23T19:07:47Z [src/client_handler.rs:685] 13eeba..: Accepted App(Bls(908ef8..), owner: Bls(b7fbc9..)) on 127.0.0.1:48663.
INFO 2019-08-23T19:07:47Z [src/client_handler.rs:152] 13eeba..: Connected to new client on 127.0.0.1:37891
TRACE 2019-08-23T19:07:47Z [src/vault.rs:188] Node(13eeba..): Succesfully sent message to: 127.0.0.1:37891
INFO 2019-08-23T19:07:47Z [src/client_handler.rs:677] 13eeba..: We already have Client(Bls(b7fbc9..)) on 127.0.0.1:47372. Cancelling the new connection from 127.0.0.1:37891.
INFO 2019-08-23T19:07:47Z [src/client_handler.rs:213] 13eeba..: Unable to deserialise challenge from 127.0.0.1:37891: invalid value: integer `35`, expected variant index 0 <= i < 3
INFO 2019-08-23T19:07:47Z [src/client_handler.rs:156] 13eeba..: Networking error: Connection was actively cancelled
INFO 2019-08-23T19:07:47Z [src/client_handler.rs:164] 13eeba..: Disconnected from client candidate on 127.0.0.1:37891
```

**Comment by bochaco** [comment]
Also one more issue spotted by CLI unit tests. When trying to query the balance of an nonexisting SafeKey I'm getting a `0` balance rather than a "SafeKey not found" error. Is this expected and decided to be like this?

**Comment by lionel1704** [comment]
Resolved in maidsafe/safe_vault#821, 5816ad4 and [this change](https://github.com/maidsafe/safe_vault/compare/abd049797c6677b6d10f8cfdd17e42ac3e9df579..cda1a9c513fc7cba7ca5fb8032c42deb1177a199).
